### DISCUSSION TITLE
Change values of dpdata.LabeledSystem() and multi frames for ‘abacus/scf’

### DISCUSSION THREAD

**Discussion by AroundPeking** [deepmodeling/dpdata#454] [General]
Two questions I wanna ask:
1. I wanna change the values of dpdata.LabeledSystem(), like that dsys_3['virials']=dsys_2['virials']-dsys_1['virials'],
   and issue showed:
![E~N{GCXE1Z7G5D)CG TL2GS](https://user-images.githubusercontent.com/108850865/233356655-af6891e3-58c2-4c88-90a1-e777dd017322.png)
 So I wanna ask how to change the values conveniently?
2. Now I have much more output of SCF calculation with "abacus/scf" format, and I wanna use 
  dsys = dpdata.LabeledSystem(dir,fmt="abacus/scf") to load and then transform them to deepmd format for training. But it seems that dpdata.LabeledSystem(dir,fmt="abacus/scf") do not support multi frames. 
How can I load them conveniently?

**Comment by wanghan-iapcm** [comment]
1. You may hack it by dsys_3.data['virials'] = dsys_2['virials']-dsys_1['virials']
2. @pxlxingliang

**Comment by njzjz** [comment]
Please see the implementation of the following method, and I guess you just need it
https://docs.deepmodeling.com/projects/dpdata/en/master/api/dpdata.html#dpdata.system.LabeledSystem.correction

**Comment by AroundPeking** [comment]
Thanks! These two methods, .data() and .correction(), can solve my first issue.

**Comment by pxlxingliang** [comment]
Hi, @AroundPeking 

For multiple frames, you can append all ` LabeledSystem` to one `LabeledSystem`, like:
https://github.com/deepmodeling/dpgen/blob/master/dpgen/generator/run.py#L4065

```
        for ii, oo in zip(sys_input, sys_output):
            _sys = dpdata.LabeledSystem(
                oo, fmt="abacus/scf", type_map=jdata["type_map"]
            )
            if len(_sys) > 0:
                if all_sys == None:
                    all_sys = _sys
                else:
                    all_sys.append(_sys)
```

**Reply by AroundPeking** [reply]
Thanks, I will try.
